es/es-server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -134,7 +135,12 @@
         /**
          * Indicates an index-level setting that is privately managed. Such a setting can not even be set on index creation.
          */
-        PrivateIndex
+        PrivateIndex,
+
+        /**
+         * Indicates a setting that contains sensitive information. Such a setting will be masked when shown to non-authorized users.
```

Comment:
Is the second sentence is relevant here? Such as it also holds for for authorized users.